pkgs/shells/nix-your-shell/default.nix

```
@@ -0,0 +1,26 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "nix-your-shell";
+  version = "v1.0.1";
+
+  src = fetchFromGitHub {
+    owner = "MercuryTechnologies";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-kdZFwMHatnhdXGSIItuE3g27qqUKqT/Hkbz13Ba5eq4=";
+  };
+
+  cargoSha256 = "sha256-dEEv0KVTPNB55ZGTsz2VpRHtmagbakd6Z4lmWab4lqw=";
+
+  meta = {
+    description = "A `nix` and `nix-shell` wrapper for shells other than `bash`";
+    homepage = "https://github.com/MercuryTechnologies/nix-your-shell";
+    license = [lib.licenses.mit];
+    maintainers = [lib.maintainers._9999years];
+  };
+
+  passthru.updateScript = ./update.sh;
```

`nix-update-script` should work here, and this can go before `meta`

```suggestion

  passthru.updateScript = nix-update-script { };
```

---

The `nix-update-script` doesn't appear to update the hashes for `fetchFromGitHub` or `cargoSha256`.

---

I might try to improve on `nix-update-script`, though, because it would be nice for it to be able to handle this (typical and fairly minimal) `buildRustPackage` invocation.